src/storage/src/controller.rs

```diff
@@ -843,15 +777,17 @@ where
             .map(|client| client.response_stream())
             .enumerate()
             .collect::<StreamMap<_, _>>();
-        if clients.is_empty() {
-            // If there are no clients, block forever. This signals that there
-            // may be more work to do (e.g., if this future is dropped and
-            // `create_collections` is called). Awaiting the stream map would
-            // return `None`, which would incorrectly indicate the completion
-            // of the stream.
-            return future::pending().await;
-        }
-        self.state.stashed_response = clients.next().await.map(|(_id, res)| res)
+
+        let msg = tokio::select! {
+            // Order matters here. We want to process internal commands
+            // before processing external commands.
+            biased;
+
+            Some(m) = self.internal_response_queue.recv() => m,
+            Some((_id, m)) = clients.next() => m,
+        };
+
+        self.state.stashed_response = Some(msg);
```

Comment:
@benesch I just made this stuff up and would love your opinion. The intent is that there is now always an `internal_response_queue` to await, so the "no clients" issue shouldn't exist. I'm not sure I'm respecting cancelation safety and such, but it looks like the `StreamMap` probably wasn't either so hoping it's ok.

Reply:
This bit LGTM! We're now mandating that the clients are cancel safe so this is all good.